server/eclipse-project/src/main/java/us/freeandfair/corla/Main.java

```diff
@@ -75,29 +97,70 @@
    */
   public Main(final /*@ non_null */ Properties the_properties) {
     my_properties = the_properties;
+    my_primary_service = Service.ignite();
+    my_redirect_service = Service.ignite();
   }
   
   // Instance Methods
   
   /**
-   * Starts a ColoradoRLA server.
+   * Parse a port number from properties.
+   * 
+   * @param the_property The name of the property.
+   * @param the_default The default port number.
    */
-  public void start() {
-    LOGGER.info("starting server with properties: " + my_properties);
+  private int parsePortNumber(final String the_property, final int the_default) {
+    int result = the_default;
     
-    // get the port number from properties
-    int port_number = DEFAULT_PORT;
     try {
-      final int prop_port_number =
-          Integer.parseInt(my_properties.getProperty("port", String.valueOf(port_number)));
-      if (MIN_PORT <= prop_port_number && prop_port_number < MAX_PORT) {
-        port_number = prop_port_number;
+      final int prop_port =
+          Integer.parseInt(my_properties.getProperty(the_property, 
+                                                     String.valueOf(the_default)));
+      if (MIN_PORT <= prop_port && prop_port < MAX_PORT) {
+        result = prop_port;
       } else {
-        LOGGER.info("invalid port number in properties, using default port");
+        LOGGER.info("invalid port number in property " + the_property + 
+                    ", using default " + the_default);
       }
     } catch (final NumberFormatException e) {
-      LOGGER.info("could not read port number from properties, using default port");
+      LOGGER.info("could not read port number property " + the_property + 
+                  ", using default " + the_default);
+    }
+    
+    return result;
+  }
+  
+  /**
+   * Redirect a request from HTTP to HTTPS.
```

Comment:
To me, the existence of this method is evidence that it would be better to configure TLS at Nginx, terminate it there, and reverse proxy. Failing that, I would suggest never serving plain HTTP at all.